### ISSUE TITLE
dmashuda-journal 

### ISSUE THREAD

**Issue by dmashuda** [awesomegroupidunno/senior-design#1]

**Comment by dmashuda** [comment]
#### Progress Report: 2015-09-21
- Worked with @numberQ to create design for client be4fe45f60d5e09c2f353273fd08ec9d55282d4c 
  - Utilization of the Adapter Design pattern: https://en.wikipedia.org/wiki/Adapter_pattern
- Started work on implementing UDP server: https://github.com/awesomegroupidunno/game-server/commit/3059a6bdabf67c24d432cb950a3de8d8b8d1cce8

**Comment by dmashuda** [comment]
## Progress Report: 2015-09-28
- converted Client UML to violet
- Worked with @numberQ  and @austin4e for refactoring UML for client
- Worked with @numberQ  and @austin4e for shared UML
- Created server UML
- https://github.com/awesomegroupidunno/senior-design/blob/master/client.png
- https://github.com/awesomegroupidunno/senior-design/blob/master/server.png
- https://github.com/awesomegroupidunno/senior-design/blob/master/shared.png
- Issues Closed:
  - #18 
  - #17 
  - #24 
  - #23 
  - #7 
  - #6 
  - #3 

TLDR;  Got UML to a good place, finished UDP receiver, created gamestate objects, created command object 

Significant Commits:
- https://github.com/awesomegroupidunno/game-server/commit/13db3c2dfabbad56e328fba5d21b4935a78318f8
- https://github.com/awesomegroupidunno/game-server/commit/7278a4c0e6b83c25abe831210be54c0bf9ac06dd
- https://github.com/awesomegroupidunno/game-server/commit/f3f3fd55cb4d7b0189f35dc7817846a88c13224b
- https://github.com/awesomegroupidunno/game-server/commit/d1f44ba90277dff912d774d21d6c35e079e22923

**Comment by dmashuda** [comment]
### 2015-10-05
- Added in parsing for subcommands
- Added threading for udp receiver and router
- Added gamemanager with threading https://github.com/awesomegroupidunno/game-server/commit/610cf3ea6f8987c7a3e5a7ae0ce39ade21ed5596
- Code cleanup https://github.com/awesomegroupidunno/game-server/commit/d87cc9356d8c5ae81c9ccbff956aa1e3433307b1 
- Fixed concurrency related issues(hooray "-race" flag) https://github.com/awesomegroupidunno/game-server/commit/b43e49ec331798526791cc18ec010558e676e2d1
- Fixed bug of the CS201 level (object passed by value when it should have been passed by reference) https://github.com/awesomegroupidunno/game-server/commit/0ed63cd4040ff554752e1c1bac4552562229d9cb

Closed issues:
- #31 
- #30 
- #25 
- #19
- #23 

#### Goals for the next week
- Create commandProcessorFactory, CommandProcessor, and implementations

**Comment by dmashuda** [comment]
## Progress Report: 2015-10-12
- Refactored server to subscriber method rather than having clients poll for updates
- Created CommandProcessorFactory
- Created ConnectCommand
- Created ConnectCommandProcessor
- Code Cleanup
- Increased Test Coverage
- Added godocs to project

### Closed Issues
- #26 
- #27 
- #20 
- #32 

### Goals For Next Week
-  TurnCommandProcessor
-  AccelerationCommandProcessor
- Add physics for motion in game loop

**Comment by dmashuda** [comment]
## Progress Report 2015-10-26
- Created  TurnCommandProcessor
- Created AccelerationCommandProcessor
- Created basic 2d physics for motion
- Refactored CommandProcessorFactory to reuse processors instead of recreating them all of the time
- Refactored Physics to be an Object with configurable options
- Implemented basic friction on vehicles
- Fixed a couple bugs with the client

### Client Commits
- Fixed SDL with OSX https://github.com/awesomegroupidunno/game-client/commit/0ae73db8ba4f8fab3eede988df11d0b51393d074
- Fixed bug with stack based allocation: https://github.com/awesomegroupidunno/game-client/commit/f23cc306c7bf6c06b8cd759a7690bba0005f2b33

### Server Commits
- Added Physics Configuration object: https://github.com/awesomegroupidunno/game-server/commit/8ca7fba6da916e8da123df606d41badcf91ca6db
  - Adjusted Physics to be reasonable: https://github.com/awesomegroupidunno/game-server/commit/f35e1c40a4eb02044d75679e8e2e7f7bf06b8ca2
  - Initial Movement: https://github.com/awesomegroupidunno/game-server/commit/275a104924d3698beea942b5505eaa6979fe3eab

### Goals for next week:
- Object Collision!

**Comment by dmashuda** [comment]
## Progress Report 2015-11-02
- Created Box2d interface for rectangular objects to detect collisions
- Wrote tests for UDP Receiver, and fixed issues found by tests 
- Fixed 2 data races 
- Performed code cleanup and fixed convention errors and increased project consistency

### Commits
- https://github.com/awesomegroupidunno/game-server/commit/b9791a8016853aa2e622dcb6132c19ecc43fb4d0 
- https://github.com/awesomegroupidunno/game-server/commit/d7456a5c4dfd1bc9f3d8176f1c4e8effa00dfe5e
- https://github.com/awesomegroupidunno/game-server/commit/1d797777a7a9b197c59b6f0fa2733ff3d616eb6a
- https://github.com/awesomegroupidunno/game-server/commit/86a68d97cbc399c58e358b0b19c8f450b119fee3

### Goals for next week:
- replace naive collision detection with proper collision detection
- work on collision physics

**Comment by dmashuda** [comment]
## Progress Report 2015-11-09
- Implemented SAT collision Detection
- Implemented Bullets on server side
- Bounded Vehicles to world bounds
- Implemented Bullet rate limiting
- Implemented Bullet cleanup

### Commits
- https://github.com/awesomegroupidunno/game-server/commit/47e32eda61e26fb0b1e9aebfcd8fa96bc11f67db
- https://github.com/awesomegroupidunno/game-server/commit/e200af812e2cadb57a02f3ad891f9d998d144246
- https://github.com/awesomegroupidunno/game-server/commit/934723ba0decc089b3e0ea4791a320b2a0754fa7
- https://github.com/awesomegroupidunno/game-server/commit/45f219c1ac6a6f48d6fa424267109095a55b46bf
- https://github.com/awesomegroupidunno/game-server/commit/51f36a0e99873281cd37951c5108fd1a3893a488

### Goals for next week:
- Add Bases and shield generator to screen
- Implement health system

**Comment by dmashuda** [comment]
## Progress Report 2015-11-16
- Implemented health system for vehicles
- Implemented health system for bases
- Implemented health system for Shield Generators
- Implemented Shield to protect bases
- Implemented Shield being disabled when generator is destroyed
- Implemented vehicle responding

### Commits
- https://github.com/awesomegroupidunno/game-server/commit/5695f15162ec256fac2c9835f41c8788c20e9308
- https://github.com/awesomegroupidunno/game-server/commit/48e224e9b0bb89e1fbfbf0ec6c52f71551047f3a
- https://github.com/awesomegroupidunno/game-server/commit/eea1fb543b1ba1f40f21592e5014bf8c440d5eff
- https://github.com/awesomegroupidunno/game-server/commit/26be6507a5f504c9e95f9d5e5fd19f711c82fcb3

### Goals for next week
- Implement power up system
- Implement a basic PowerUp

**Comment by dmashuda** [comment]
## Progress Report: 2015-11-23
- Refactored position and sizing system
- Added base system for powerups
- Added powerup game objects
- added powerup command to use powerups
- added powerup command processor
- added healing powerup
- added speed boost powerup
- added base for seeking rocket

### Commits
- https://github.com/awesomegroupidunno/game-server/commit/244744d1876ab80375d8f5421807403595c56abe
- https://github.com/awesomegroupidunno/game-server/commit/f5a0850470a57b02dc0ae8badf12a3f1ebcc037e

### Goals for next week
- complete rocket powerup
- increase test coverage
- Refactoring

**Comment by dmashuda** [comment]
## Progress Report: 2015-11-30
- Built function to find an empty space in the game world
- Added basic gravity bullets, disabled by default
- Added random powerup spawning
- Added reliable vehicle spawning
- Added Rocket targeting

## Commits
- https://github.com/awesomegroupidunno/game-server/commit/9562669d897db484ec746361196dd427d6ca67fa
- https://github.com/awesomegroupidunno/game-server/commit/6b6f473ed2d3215ceb7c4591d1cf9950dd73dec5

### Goals for next week
- Complete Rocket powerup
- Increase test coverage

**Comment by dmashuda** [comment]
## Progress Report 2015-12-07
- Finished Rocket powerup on the server side
- Refactored physics code to cut down on duplicate code
- Implemented gravity well powerup

#### Commits
- https://github.com/awesomegroupidunno/game-server/commit/96b9c33904a1c0204eefbd2de7829f45b99d0e7f
- https://github.com/awesomegroupidunno/game-server/commit/4c1b0d67e4bca3febbe1fea8cffa47a6547bc388
- https://github.com/awesomegroupidunno/game-server/commit/7ef69282f2a6830ea06656deb6b98efe3dca20a9

### Goals for next week
- Fix any bugs that come up
- Write Paper!

**Comment by dmashuda** [comment]
## Progress Report 2015-12-14
- The game now restarts after gameover!
- Worked on paper
